tests/conftest.py

```diff
@@ -271,3 +272,14 @@ def running_clients(hlwm, running_clients_num):
     "running_clients_num" test parameter.
     """
     return hlwm.create_clients(running_clients_num)
+
+
+@pytest.fixture()
+def keyboard():
+    class KeyBoard:
+        def press(self, s):
+            subprocess.call('xdotool key x'.split())
+            # Bad workaround for keypress injection not being synchronous:
+            time.sleep(.5)
```

Comment:
This is a really bad hack and I'm not sure if we want to merge this. Any ideas on how to solve it properly?

Reply:
We can read herbstluftwm's stderr and wait for the keypress to be processed. Shouldn't this be `s` instead of `x` in the xdotool-call?

Reply:
What we can do in this (test-)case: run a `read -n 1` in the terminal, type the `x` and expect the terminal to close. (We can make a waitpid with a timeout of 5 seconds so we only wait the full 5 seconds if it fails).

Reply:
Yeah, I forgot to actually use the parameter... will fix it.

So, there is no generic way of checking whether a keypress has been processed by *any* client, right? Is there some X queue that we could inspect?

Reply:
I when with the `read -n 1` solution. PRs #305 and #304 are included and need to be merged first.